docs/design/gatewayapi-translator.md

```diff
@@ -0,0 +1,242 @@
+# Gateway API Translator Design
+
+## Assumptions
+- initially target core conformance features only, to be followed by extended conformance features
+
+## Inputs and Outputs
+
+The main inputs to the Gateway API translator are:
+- the GatewayClass to process
+- Gateways, HTTPRoutes, Services, Secrets
```

Comment:
Perhaps "in addition to", the details of the endpoint resolver & how it integrates with the rest of the translator still need to be worked out. Services are still required in order to do proper validation of the Gateway API backend refs, though.

Reply:
do you think the info in https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/ is enough to validate the Gateway API backend ref ?

Reply:
I'm not sure what you're trying to achieve with this optimization @arkodg? The Gateway API assumes that backendRefs will be Services, for the most part. Yes, that means that the translator will need to go and look up EndpointSlices or Endpoints, but that's an implementation detail.

Reply:
In Contour, for example, the Endpoint translator is actually separate to the others because it changes so much more often. tbh I'm not sure if this turned out to be a good idea or not though.

Reply:
in the current [design](https://github.com/envoyproxy/gateway/blob/main/docs/design/SYSTEM_DESIGN.md) the resource provider provides the resources to the translator rather than the resource translator watching for endpoints. I was hoping the doc would outline all the resource types needed to build out the IR.

Reply:
> do you think the info in https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/ is enough to validate the Gateway API backend ref ?

Similar to what @youngnick said, Gateway API backedRefs are expected to be to Services so I think we need to validate that those Services exist, then can subsequently use their Endpoint data. 

 > In Contour, for example, the Endpoint translator is actually separate to the others because it changes so much more often. tbh I'm not sure if this turned out to be a good idea or not though.

Yeah, I was actually under the impression we were planning to do the same for envoy-gateway (I thought we had a separate service resolver component), but I probably missed some updates. If the plan is to have Endpoints go through the Gateway API translator like all the other resources, then yes, they'll need to be added as required inputs in addition to Services.